packages/vaadin-grid/src/iron-list.js

```diff
@@ -588,6 +588,15 @@ export const PolymerIronList = Class({
    * @param {!Array<number>=} itemSet
    */
   _updateMetrics: function (itemSet) {
+    // Sometimes the call for this method is schedule before Grid is hidden
+    // but it happens when Grid is already hidden.
+    // When this happens, there might be some issues as described
+    // here https://github.com/vaadin/web-components/issues/2127.
+    // Skipping any calculation for a hidden Grid.
+    if (this.hidden) {
```

Comment:
Here, I could also use `_isVisible`, which does `return Boolean(this.offsetWidth || this.offsetHeight)`. Both works, but `_isVisible` is used in other places on the class, while `hidden` is not.

```suggestion
    if (!this._isVisible) {
```